java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S5852_java.html

```diff
@@ -14,21 +14,55 @@ <h2>Ask Yourself Whether</h2>
+  and <code>.*_.*</code> have quadratic runtime. </li>
+  <li> If you're performing a partial match (such as by using <code>Matcher.find</code>, <code>String.split</code>, <code>String.replaceAll</code>
+  etc.) and the regex is not anchored to the beginning of the string, quadratic runtime is especially hard to avoid because whenever a match fails,
+  the regex engine will try again starting at the next index. This means that any unbounded repetition (even a possessive one), if it's followed by a
+  pattern that can fail, can cause quadratic runtime on some inputs. For example <code>str.split("\\s*,")</code> will run in quadratic time on strings
+  that consist entirely of spaces (or at least contain large sequences of spaces, not followed by a comma). </li>
+</ul>
+<p>In order to rewrite your regular expression without these patterns, consider the following strategies:</p>
+<ul>
+  <li> If applicable, define a maximum number of expected repetitions using the bounded quantifiers, like <code>{1,5}</code> instead of <code>+</code>
+  for instance. </li>
   <li> Refactor <code>nested quantifiers</code> to limit the number of way the inner group can be matched by the outer quantifier, for instance this
-  nested quantifier situation <code>(ba+)+</code> doesn't cause performance issues, indeed, the inner group can be matched only if it exists exactly
-  one <code>b</code> char per repetition of the group. </li>
-  <li> Make sure that consecutive repetitions can't match the same contents or are separated by a non-optional separator that can't be matched by both
-  of the repetitions. So for example <code>a*b*</code> is not a problem because <code>a*</code> and <code>b*</code> match different things and
-  <code>a*_a*</code> is not a problem because the repetitions are separated by a '_' and can't match that '_'. However, <code>a*a*</code> and
-  <code>.*_.*</code> have quadratic runtime. </li>
+  nested quantifier situation <code>(ba+)+</code> doesn't cause performance issues, indeed, the inner group can be matched only if there exists
+  exactly one <code>b</code> char per repetition of the group. </li>
   <li> Optimize regular expressions with <code>possessive quantifiers</code> and <code>atomic grouping</code>. </li>
   <li> Use negated character classes instead of <code>.</code> to exclude separators where applicable. For example the quadratic regex
   <code>.*_.*</code> can be made linear by changing it to <code>[^_]*_.*</code> </li>
 </ul>
+<p>Sometimes it's not possible to rewrite the regex to be linear while still matching what you want it to match. Especially when using partial
+matches, for which it is quite hard to avoid quadratic runtimes. In those cases consider the following approaches:</p>
+<ul>
+  <li> Solve the problem without regular expressions </li>
+  <li> Use an alternative non-backtracking regex implementations such as Google's <a href="https://github.com/google/re2">RE2</a> / <a
```

Comment:
We should replace the `/` sign with an `and` or `or` as to not confuse `RE2 and RE2/J` for `RE2 and RE2 and J`